BlueDBOnDisk/src/main/java/org/bluedb/disk/collection/FacadeCollection.java

```diff
@@ -38,8 +39,19 @@ public FacadeCollection(ReadableBlueDbOnDisk db, String name, Class<T> valueType
 	}
 
 	@Override
-	public <K extends ValueKey> BlueIndex<K, T> getIndex(String name, Class<K> keyType) throws BlueDbException {
-		return getCollection().getIndex(name, keyType);
+	public <K extends ValueKey> BlueIndex<K, T> getIndex(String indexName, Class<K> indexKeyType) throws BlueDbException {
+		return new FacadeBlueIndexOnDisk<K, T>(() -> {
+			ReadableBlueCollection<T> collection = getCollection();
+			if (collection instanceof DummyReadOnlyBlueCollectionOnDisk) {
+				return null;

```

Comment:
Isn't it supposed to return a dummy index here?

Reply:
The Supplier gets one if it can, if not, it returns null and the FacadeCollection creates a Dummy.  Alternatively, we could create a dummy here.  I see pros and cons either way.

Reply:
This method is something that a user would call. Returning null will give them null right? Unless I'm missing something. Shouldn't we just return dummyCollection.getIndex() instead of null?

Reply:
Oh sorry, we're returning the facade index here. And the facade index returns a dummy if this returns null. Got it